Cleanup and fix management commands.

vil-s opened the pull request:
- Ensure that new switch is not created unless the --create flag is passed.
- Add tests to ensure no regression for the --create fix on all commands.
- Add separate positional arguments for commands (instead of "positionals") to improve the --help documentation.

clintonb commented on waffle/management/commands/waffle_flag.py:
```diff
@@ -8,74 +8,76 @@
             dest='list_flags',
             default=False,
-            help="List existing samples."),
+            help="List existing samples.")
```
Single quotes

clintonb commented on waffle/management/commands/waffle_flag.py:
```diff
@@ -8,74 +8,76 @@
             action='store_true',
             dest='everyone',
-            help="Activate flag for all users."),
+            help="Activate flag for all users.")
```
Single quotes

clintonb commented on waffle/management/commands/waffle_flag.py:
```diff
@@ -8,74 +8,76 @@
             action='store_false',
             dest='everyone',
-            help="Deactivate flag for all users."),
+            help="Deactivate flag for all users.")
```
Single quotes

clintonb commented on waffle/management/commands/waffle_sample.py:
```diff
@@ -51,7 +60,7 @@ def handle(self, *args, **options):
                 sample = Sample.objects.get(name=sample_name)
             except Sample.DoesNotExist:
-                raise CommandError('This sample does not exist.')
+                raise CommandError("This sample doesn't exist.")
```
Single quotes. I find "does not" to be more understandable (esp. for non-native English speakers) than "doesn't".

clintonb commented on waffle/tests/test_management.py:
```diff
@@ -28,6 +28,15 @@ def test_create(self):
+        """ The command shouldn't create a new flag if the create flag is
+        not set.
+        """
+        with self.assertRaisesRegexp(CommandError, "This flag doesn't exist."):
```
Applicable to all of your new tests:

1. Use single quotes to maintain consistency.
2. Add an assertion that actually ensures the flag/switch/sample does not exist.

jsocol replied:
Fwiw, I strongly prefer double quotes when the string has apostrophes in it, to avoid awkward escape sequences:

```py
'don\'t do this if it can be avoided'
"this's better"
'but this is best'
```

clintonb reviewed (changes requested):
Please squash your commits.

clintonb commented on waffle/tests/test_management.py:
```diff
@@ -28,6 +28,17 @@ def test_create(self):
+        not set.
+        """
+        name = 'test'
+        self.assertFalse(Flag.objects.filter(name=name).exists())
```
The assertions should come _after_ the calls to the management command. The purpose of the test is to ensure the command doesn't create a new instance.

vil-s replied:
aha, I misunderstood your previous comment

clintonb reviewed (changes requested):
Correct the tests.

Commit:
Cleanup and fix management commands.
- Ensure that new switch is not created unless the --create flag is passed.
- Add tests to ensure no regression for the --create fix on all commands.
- Add separate positional arguments for commands (instead of "positionals") to improve the --help documentation.
